eclipse-theia/theia
# Apply CommentThread vscode Plugin API

#### What it does

Apply [CommentThread vscode Plugin API](https://code.visualstudio.com/api/references/vscode-api#comments)

related CQ: http://dev.eclipse.org/ipzilla/show_bug.cgi?id=22902

fixes https://github.com/eclipse-theia/theia/issues/8492
#### How to test

1. Remove `"@theia/git": "^1.8.0"` from `examples/browser/package.json` file.
2. Download [vscode-git-1.49.3](https://open-vsx.org/api/vscode/git/1.49.3/file/vscode.git-1.49.3.vsix) and copy it to the `plugins` folder.
3. Download [vscode GitHub Pull request plugin](https://github.com/microsoft/vscode-pull-request-github/releases/download/0.20.0/vscode-pull-request-github-0.20.0.vsix) and copy it to the `plugins` folder.
4. Apply GitHub authentication provider:
* Git clone the [github authentication plugin](https://github.com/vinokurig/github-authentication-plugin.git) to the `plugins` folder.
* Paste [your GitHub token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) to the plugin [here](https://github.com/vinokurig/github-authentication-plugin/blob/89b951d1a4280aee62635f5fa2b99feaf92aeacb/src/github-authentication-plugin.ts#L26).
* Compile the plugin. Note that errors might appear in the compilation log but it doesn't prevent the plugin from working properly.
5. Start Theia and  open a project cloned from GitHub with a Pull-Request e.g. https://github.com/vinokurig/environment-variable.git
6. Sign in the `GitHub Pull-Requests Plugin`:
![screenshot-serverg0cevid1-che-dev-server-3010 192 168 99 253 nip io-2020 09 23-11_42_23](https://user-images.githubusercontent.com/7668752/93988856-e1e22180-fd91-11ea-8cfb-9fc2076f2243.png)
7. Open a pull-request in the plugin: 
![screenshot-localhost_3000-2020 12 16-12_13_45](https://user-images.githubusercontent.com/7668752/102335091-20a20500-3f98-11eb-99a6-a9b3635cd5d4.png)
8. Open a changed file with comments: 
![screenshot-localhost_3000-2020 12 16-12_16_21](https://user-images.githubusercontent.com/7668752/102335378-7a0a3400-3f98-11eb-99aa-8e2afe6744ba.png)

See: Pull-Request comments are available in the diff. You can add some new comments by clicking the white line and delete and edit existing comments.
#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
